app/models/patron.rb

```diff
@@ -5,24 +5,25 @@
 class Patron
   attr_reader :record
 
-  def self.find_by(library_id: nil, patron_key: nil)
-    return if library_id.nil? && patron_key.nil?
+  # rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+  def self.find_by(sunetid: nil, library_id: nil, patron_key: nil, symphony_client: SymphonyClient.new)
+    patron_key ||= symphony_client.login_by_sunetid(sunetid)&.dig('key') if sunetid
+    patron_key ||= symphony_client.login_by_library_id(library_id)&.dig('key') if library_id
 
-    symphony_client = SymphonyClient.new
-
-    patron_key ||= symphony_client.login_by_library_id(library_id)&.dig('key')
-
-    return if patron_key.blank?
-
-    new(symphony_client.patron_info(patron_key))
+    new(symphony_client.patron_info(patron_key)) if patron_key.present?
```

Comment:
Do we want to notify Honeybadger or something like that if we run into the case of a patron w/o a key (to see how often this is actually happening)?